CriteoPublisherSdk/Sources/Util/CR_URLResolver.m

```diff
@@ -102,12 +111,16 @@ - (void)URLSession:(NSURLSession *)session
                     task:(NSURLSessionTask *)task
     didCompleteWithError:(nullable NSError *)error {
   if (error != nil) {
-    self.resolutionHandler([CR_URLResolution resolutionError:error]);
+    CR_URLResolution *resolution = [CR_URLResolution resolutionError:error];
+    [self resolveWithResolution:resolution];
     return;
   }
-  CR_URLResolution *resolution = [CR_URLResolution resolutionWithType:CR_URLResolutionStandardUrl];
-  resolution.URL = task.currentRequest.URL;
-  self.resolutionHandler(resolution);
+
+  if (!self.resolved) {
+    CR_URLResolution *resolution = [CR_URLResolution resolutionWithType:CR_URLResolutionStandardUrl
+                                                                    URL:task.currentRequest.URL];
+    self.resolutionHandler(resolution);
```

Comment:
Souldn't you call the new `resolveWithResolution` method ?

Reply:
Fixed